packages/inter-protocol/src/psm/psm.js

```diff
@@ -243,47 +247,75 @@ export const start = async (zcf, privateArgs, baggage) => {
     E(anchorBrand).getAmountShape(),
     E(stableBrand).getAmountShape(),
   ]);
-  const makeWantStableInvitation = () =>
-    zcf.makeInvitation(
-      wantStableHook,
-      'wantStable',
-      undefined,
-      M.split({ give: { In: anchorAmountShape } }),
-    );
-  const makeGiveStableInvitation = () =>
-    zcf.makeInvitation(
-      giveStableHook,
-      'giveStable',
-      undefined,
-      M.split({ give: { In: stableAmountShape } }),
-    );
 
-  const publicFacet = Far('Parity Stability Module', {
-    getMetrics: () => metricsSubscriber,
-    getPoolBalance: () => anchorPool.getAmountAllocated('Anchor', anchorBrand),
-    makeWantStableInvitation,
-    makeGiveStableInvitation,
+  const PSMI = M.interface('PSM', {
+    getMetrics: M.call().returns(M.remotable('MetricsSubscriber')),
+    getPoolBalance: M.call().returns(anchorAmountShape),
+    makeWantStableInvitation: M.call().returns(M.promise()),
+    makeGiveStableInvitation: M.call().returns(M.promise()),
+    ...publicMixinAPI,
   });
 
-  const getRewardAllocation = () => feePool.getCurrentAllocation();
+  const publicFacet = vivifyFarInstance(
+    baggage,
+    'Parity Stability Module',
+    PSMI,
+    {
+      getMetrics() {
+        return metricsSubscriber;
+      },
+      getPoolBalance() {
+        return anchorPool.getAmountAllocated('Anchor', anchorBrand);
+      },
+      makeWantStableInvitation() {
+        return zcf.makeInvitation(
+          wantStableHook,
+          'wantStable',
+          undefined,
+          M.split({
+            give: { In: anchorAmountShape },
+            want: M.or({ Out: stableAmountShape }, {}),
+          }),
+        );
+      },
+      makeGiveStableInvitation() {
+        return zcf.makeInvitation(
+          giveStableHook,
+          'giveStable',
+          undefined,
+          M.split({
+            give: { In: stableAmountShape },
+            want: M.or({ Out: anchorAmountShape }, {}),
+          }),
+        );
+      },
+      ...publicMixin,
+    },
+  );
+
+  // TODO why does this operation return an object with a single operation?
```

Comment:
It gets used in places where objects can be passed, but functions cannot.  If that's no longer true, it can be simplified, I think.